packages/cli/generators/controller/templates/src/controllers/controller-crud-template.ts

```diff
@@ -0,0 +1,39 @@
+import {Filter, Where} from '@loopback/repository';
+import {inject} from '@loopback/context';
+import {<%= modelName %>} from '../models';
+import {<%= repositoryName %>} from '../repositories';
+
+export class <%= name %>Controller {
+  @inject('repositories.<%= modelNameCamel %>') <%= repositoryNameCamel %> : <%= repositoryName %>;
```

Comment:
IMO, we should be using constructor injection, because we don't have any reasonable default to use for this repository if the user of our API does not provide us any. We are advocating for this approach in our best practices too, see http://loopback.io/doc/en/lb4/Implementing-features.html#decouple-controller-from-repository

Further reading: https://social.msdn.microsoft.com/Forums/office/en-US/7a1dbdd5-ddf2-4eb2-8063-a423bb441158/when-to-use-property-injection-vs-constructor-injection?forum=csharpgeneral

> If the class itself has a strong dependency upon the interface/object that you inject it with, you should use constructor injection as the class is pretty much useless without the dependency, i.e. it makes no sense to instantiate it without it. A constructor therefore enforces the dependency requirement.
>
> But if it makes sense for the class to be able to do its job without using the dependency, you could use property injection. When using property injection the dependency may or may not used depending on whether the property is actually invoked.
>
> So it depends.
>
> Please refer to the following threads for more information about this:
http://stackoverflow.com/questions/1503584/dependency-injection-through-constructors-or-property-setters
http://stackoverflow.com/questions/21218868/setter-injection-vs-constructor-injection
http://programmers.stackexchange.com/questions/300706/dependency-injection-field-injection-vs-constructor-injection